python/hsfs/core/job_configuration.py

```diff
@@ -43,8 +44,15 @@ def __init__(
         self._dynamic_allocation = dynamic_allocation
         self._dynamic_min_executors = dynamic_min_executors
         self._dynamic_max_executors = dynamic_max_executors
+        self._properties = properties
 
     def to_dict(self):
+        default_property = "spark.yarn.maxAppAttempts=2"
+        if not self._properties:
+            self._properties = default_property
+        else:
+            self._properties = self._properties + (f"\n{default_property}" if default_property.split("=")[0] not in self._properties else "")
```

Comment:
I think this code would be more readable if instead of having 1 constant with `key=value` we would have 2 constants, one with the key and one with the value - you wouldn't need to split it here.